### PR TITLE
fix: dapp-issues

### PR DESCRIPTION
**Pull request by samobasquiat**
- [ ] Tooltips should be confined within the widget boundaries: 
image image
- [ ] Is it possible to utilize our collapsible component with animation? 
-  image
- [ ] The hover styles for the list item component are inconsistent as they remove the bottom border. 
 image
- [ ] We should only accept and show numerical values. -> 2
-
 image
- [ ] The ripple animation effect should be confined to the button's boundary. When clicking on the back button, the color of the cancel text changes
- [ ] Replace theme icons with the new ones
- [ ] infinite approval icon
- [ ] modal open transition issue
- [ ] wallets on mobile screens
- [ ] add failed message to failed modal desc

# Checklist:

- [x] I have performed a self-review of my code
- [x] Implemented a user interface (UI) change, referencing our Figma design to ensure pixel-perfect precision.

### CODE REVIEW

File: widget/embedded/src/components/Slippage/Slippage.tsx [hunk 1]
```diff
@@ -69,6 +69,12 @@ export function Slippage() {
           variant="contained"
           value={customSlippage || ''}
           color="dark"
+          onKeyDown={(event) => {
+            const disallowedKeys = ['-', '+'];
+            if (disallowedKeys.includes(event.key)) {
+              event.preventDefault();
+            }
+          }}
```
Comment (Ikari-Shinji-re) [inline widget/embedded/src/components/Slippage/Slippage.tsx:77]: It's not within your assigned task. We're encountering a similar problem with the token amount input on the homepage. Could you address that as well?

File: widget/embedded/src/components/SwapDetailsAlerts/SwapDetailsAlerts.styles.ts [hunk 2]
```diff
@@ -5,3 +5,27 @@ export const Alerts = styled('div', {
...
+  maxHeight: '0px',
+  overflow: 'hidden',
+  variants: {
+    open: {
+      true: {
+        maxHeight: '$40',
+      },
+    },
+  },
+});
```
Comment (Ikari-Shinji-re) [inline widget/embedded/src/components/SwapDetailsAlerts/SwapDetailsAlerts.styles.ts:31]: Can we try using the collapsible component here? If that's not an option, let's at least keep the transition duration consistent for uniformity : 0.5 s -> 300ms

File: widget/ui/src/components/Button/Button.styles.tsx [hunk 3]
```diff
@@ -549,10 +549,9 @@ const ripple = keyframes({
 
 export const RippleContainer = styled('div', {
   position: 'absolute',
-  top: '0',
-  right: '0',
-  bottom: '0',
-  left: '0',
+  inset: '0',
+  overflow: 'hidden',
+  borderRadius: '100%',
```
Comment (Ikari-Shinji-re) [inline widget/ui/src/components/Button/Button.styles.tsx:554]: The ripple effect isn't functioning correctly :
image01

File: widget/ui/src/components/ListItem/ListItem.styles.ts [hunk 4]
```diff
@@ -50,13 +50,15 @@ export const BaseListItem = styled('li', {
...
         [`.${darkTheme} &`]: {
-          $$color: '$colors$neutral400',
+          borderColor: '$neutral400',
         },
-        borderColor: '$$color',
         borderBottomRightRadius: 0,
         borderBottomLeftRadius: 0,
+        '&:hover': {
+          borderRadius: '$xs',
+        },
```
Comment (Ikari-Shinji-re) [inline widget/ui/src/components/ListItem/ListItem.styles.ts:61]: The top and bottom border radius don't match each other :
image02

File: widget/embedded/src/components/SwapDetailsAlerts/SwapDetailsAlerts.styles.ts (hunk 2 above)
Comment (samobasquiat) [reply widget/embedded/src/components/SwapDetailsAlerts/SwapDetailsAlerts.styles.ts]: I had tried to use it but I couldn't. Just changed the transition.

File: widget/ui/src/components/ListItem/ListItem.styles.ts (hunk 4 above)
Comment (samobasquiat) [reply widget/ui/src/components/ListItem/ListItem.styles.ts:61]: It's matched with Figma. Let's talk about it if you have any comments.

**Comment by RanGojo** [comment]
- [ ] Please check refresh icon tooltip position in expanded scenario. It seems it doesn't have correct position.
- [ ] Please rebase with next after mergin hover PR to double check.
- [ ] I should double check these two bullets with you:
 - wallets on mobile screens
 - add failed message to failed modal desc

**Review by RanGojo — CHANGES_REQUESTED** [review]
Thanks for the PR. 

- [ ] Please check refresh icon tooltip position in expanded scenario. It seems it doesn't have correct position.
- [ ] Please rebase with next after mergin hover PR to double check.
- [ ] I should double check these two bullets with you:
- wallets on mobile screens
- add failed message to failed modal desc

**Comment by RanGojo** [comment]
Please test failed swap message with this text:
`  stepMessage.detailedMessage.content = 'insufficient funds for intrinsic transaction cost | See: https://links.ethers.org/v5-errors-INSUFFICIENT-FUNDS | (error={"code":-32603,"message":"Insufficient funds for gas x price + value","data":{"originalError":{"message":"Insufficient funds for gas x price + value","stack":"Error: Insufficient funds for gas x price + value\n    at Yp.ethSendTransaction (chrome-extension://lgmpcpglpngdoalbgeoldeajfclnhafa/background.js:180:190647)"}}}, method="sendTransaction", transaction=undefined, code=INSUFFICIENT-FUNDS, version=providers/5.7.2)'`

and make sure it's correctly displayed. 

Screenshot from 2024-04-07 12-26-45

**Comment by RanGojo** [comment]
I'm sure it's a ui problem from before, but could we fix it here? we need to align wallet icons better in this page. (please make sure it doesn't break style in confirm wallets page)

Screenshot from 2024-04-07 12-30-22

**Review by Ikari-Shinji-re — APPROVED** [review]

**Comment by RanGojo** [comment]
In this case, the first sentence is not distinguished well from the error message.
Also I was thinking for removing tooltip in modal for error as we are not truncating it. 
 
Screenshot from 2024-04-08 10-01-29

Also, for the case of alert, it's better now. I'm not sure how we could further improved it. for example by having flexible expandable area height.
Screenshot from 2024-04-08 09-55-09

**Review by RanGojo — APPROVED** [review]
Well done.

**Pull request merged**